src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,92 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { RootState } from '../../app/store';
+
+type Props = {
+  updateErrorMessage: (error: string) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ updateErrorMessage }) => {
+  const dispatch = useAppDispatch();
+  const { currentTodo } = useAppSelector<RootState>(state => state);
+  const [user, setUser] = useState<User>();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const handleDeleteCurrent = () => {
+    dispatch(currentTodoActions.removeTodo());
+  };
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      updateErrorMessage('Request timed out');
+      setIsLoading(false);
+    }, 3000);
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(userData => {
+          setUser(userData);
+        })
+        .catch(error => {
+          updateErrorMessage(`Error fetching user data. ${error}`);
+        })
+        .finally(() => {
+          clearTimeout(timeoutId);
+          setIsLoading(false);
+        });
+    }
+  }, [currentTodo]);
```

Comment:
Fix linter 
<img width="1135" alt="image" src="https://github.com/mate-academy/react_redux-list-of-todos/assets/64750958/836910fb-76c5-496a-9a96-649a8af0a4f9">